cli/lsp/tsc.rs

```diff
@@ -1141,14 +1141,23 @@ impl Classifications {
       let start_pos = line_index.position_tsc(offset.into());
       let end_pos = line_index.position_tsc(TextSize::from(offset + length));
 
-      // start_pos.line == end_pos.line is always true as there are no multiline tokens
-      builder.push(
-        start_pos.line,
-        start_pos.character,
-        end_pos.character - start_pos.character,
-        token_type,
-        token_modifiers,
-      );
+      if start_pos.line == end_pos.line
```

Comment:
This directly deals with eliminating a panic that shouldn't occur, but did for me.  I haven't been able to reproduce it, but at least we will get a more useful error logged to the debug output versus a panic and a crash of the language server.

Reply:
Seems risky to just log here because maybe it could cause some indirect bugs. It might be better if we panic, but with the information you've added because then we'll get bug reports to fix the issue and we'll know this is occurring.

Reply:
I don't think it can cause indirect bugs.  What is occurring here is a returned value from tsc is being "remapped" to something that can be used by a language server client.  If it errors, a "bad" re-map is simply elided, which means a highlighted semantic symbol doesn't get returned to the client.  That is all.

Panicing here causes the language server to terminate, and be restarted which is far more "destructive".  Generally we should avoid panicing in the language server, as no good comes from it.

An improvement is I can return a result here as an internal error.